File: LuoGu/P8873.py
def check(x, k):
    if ((x+1)*(x*2+1) >= k):
        return True
    return False
for _ in range(int(input())):
    k = int(input())
    lo = 0; hi = k
    while (lo <= hi):
        mid = (lo + hi) // 2
        if (check(mid,k)):
            hi = mid - 1
        else:
            lo = mid + 1
    k -= lo*((lo-1)*2+1)
    if (k >= 3*lo + 1):
        print(k-lo*4-1)
    elif (k >= 2*lo + 1):
        print(2*lo+1 - k)
    elif (k > lo):
        print(2*lo+1 - k)
    else:
        print(k - 1)

File: LuoGu/P8873.cpp
#include <bits/stdc++.h>
using namespace std;
using LL = long long;
bool check(LL x,LL k){
    if ((x*2+1) >= 1.0*k/(x+1)) return 1;
    return 0;
}
int main(){
    int T;
    cin >> T;
    while (T--){
        LL k;
        cin >> k;
        LL lo = 0, hi = k;
        while (lo <= hi){
            LL mid = lo + hi >> 1;
            if (check(mid,k)) hi = mid - 1;
            else lo = mid + 1;
        }
        k -= lo*((lo-1)*2+1);
        if (k >= 3*lo + 1){
            cout << k-lo*4-1 << '\n';
        } else if (k >= 2*lo + 1){
            cout << 2*lo+1 - k << '\n';
        } else if (k > lo){
            cout << 2*lo+1 - k << '\n';
        } else{
            cout << k - 1 << '\n';
        }
    }
}